### PR TITLE
egui: auto save

### PR DESCRIPTION
**Pull request by steverusso**

### CODE REVIEW

File: clients/egui/src/account/mod.rs [hunk 1]
```diff
@@ -99,6 +133,23 @@ impl AccountScreen {
...
+                        }
+                    }
+                }
+                AccountUpdate::SaveResult(id, result) => match result {
+                    Ok(time_saved) => {
+                        if let Some(tab) = self.workspace.get_mut_tab_by_id(id) {
+                            tab.last_saved = time_saved;
+                        }
+                    }
+                    Err(_err) => {}
```
Comment (Parth) [inline clients/egui/src/account/mod.rs:151]: intentional?

File: clients/egui/src/account/mod.rs [hunk 2]
```diff
@@ -29,38 +32,69 @@ pub struct AccountScreen {
...
     update_rx: mpsc::Receiver<AccountUpdate>,
 
+    save_req_tx: mpsc::Sender<SaveRequest>,
+
     tree: FileTree,
     sync: SyncPanel,
     usage: Result<Usage, String>,
     workspace: Workspace,
     modals: Modals,
+    shutdown: Option<AccountShutdownProgress>,
```
Comment (Parth) [inline clients/egui/src/account/mod.rs:42]: potentially could just be a bool?

**Review by Parth — APPROVED** [review]

File: clients/egui/src/account/mod.rs (hunk 2 above)
Comment (steverusso) [reply clients/egui/src/account/mod.rs:42]: There'll be other things we add that we'd want to track on shutdown, for example, if there's a sync currently happening, finish up, etc.

File: clients/egui/src/account/mod.rs (hunk 1 above)
Comment (steverusso) [reply clients/egui/src/account/mod.rs]: Good catch, https://github.com/lockbook/lockbook/pull/1747/commits/8f72edc6e3f9e60015bca36f59bf6b80720acf2a

**Pull request merged**